x-pack/legacy/plugins/ml/server/routes/job_service.ts

```diff
@@ -0,0 +1,609 @@
+      try {
+        const { getAllGroups } = jobServiceProvider(context.ml!.mlClient.callAsCurrentUser);
+        const resp = await getAllGroups();
+
+        return response.ok({
+          body: resp,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup JobService
+   *
+   * @api {post} /api/ml/jobs/update_groups
+   * @apiName UpdateGroups
+   * @apiDescription Updates 'groups' property of an anomaly detection job
+   */
+  router.post(
+    {
+      path: '/api/ml/jobs/update_groups',
+      validate: {
+        body: schema.object(updateGroupsSchema),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const { updateGroups } = jobServiceProvider(context.ml!.mlClient.callAsCurrentUser);
+        const { jobs } = request.body;
+        const resp = await updateGroups(jobs);
+
+        return response.ok({
+          body: resp,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup JobService
+   *
+   * @api {get} /api/ml/jobs/deleting_jobs_tasks
+   * @apiName DeletingJobTasks
+   * @apiDescription Gets the ids of deleting anomaly detection jobs
+   */
+  router.get(
+    {
+      path: '/api/ml/jobs/deleting_jobs_tasks',
+      validate: false,
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const { deletingJobTasks } = jobServiceProvider(context.ml!.mlClient.callAsCurrentUser);
+        const resp = await deletingJobTasks();
+
+        return response.ok({
+          body: resp,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup JobService
+   *
+   * @api {post} /api/ml/jobs/jobs_exist
+   * @apiName JobsExist
+   * @apiDescription Checks if each of the jobs in the specified list of IDs exist
+   */
+  router.post(
+    {
+      path: '/api/ml/jobs/jobs_exist',
+      validate: {
+        body: schema.object(jobIdsSchema),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const { jobsExist } = jobServiceProvider(context.ml!.mlClient.callAsCurrentUser);
+        const { jobIds } = request.body;
+        const resp = await jobsExist(jobIds);
+
+        return response.ok({
+          body: resp,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup JobService
+   *
+   * @api {get} /api/ml/jobs/new_job_caps/:indexPattern
+   * @apiName NewJobCaps
+   * @apiDescription Retrieve the capabilities of fields for indices
+   */
+  router.get(
+    {
+      path: '/api/ml/jobs/new_job_caps/{indexPattern}',
+      validate: {
+        params: schema.object({ indexPattern: schema.string() }),
```

Comment:
you might need to add a validation schema for `query`

Reply:
Good catch - added in 1d28a85bd374085ce5f775b543268b8ab0497546